spec_42.txt

```diff
@@ -0,0 +1,179 @@
+
+* To allow the project to develop faster and more accurately, by increasing the diversity of the decision making process;
+
+* To support the natural life cycle of project versions from experimental through to stable, by allowing safe experimentation, rapid failure, and isolation of stable code;
+
+* To reduce the internal complexity of project repositories, thus making it easier for Contributors to participate and reducing the scope for error;
+
+* To enforce collective ownership of the project, which increases economic incentive to Contributors and reduces the risk of hijack by hostile entities.
+
+++ Design
+
++++ Preliminaries
+
+* The project SHALL use the git distributed revision control system.
+
+* The project SHALL be hosted on github.com or equivalent, herein called the "Platform".
+
+* The project SHALL use the Platform issue tracker.
+
+* The project SHOULD have clearly documented guidelines for code style.
+
+* A "Contributor" is a person who wishes to provide a patch, being a set of commits that solve some clearly identified problem.
+
+* A "Maintainer" is a person who merges patches to the project. Maintainers are not developers; their job is to enforce process.
+
+* Contributors SHALL NOT have commit access to the repository unless they are also Maintainers.
+
+* Maintainers SHALL have commit access to the repository.
+
+* Everyone, without distinction or discrimination, SHALL have an equal right to become a Contributor under the terms of this contract.
+
++++ Licensing and Ownership
+
+* The project SHALL use a share-alike license, such as the GPLv3 or a variant thereof (LGPL, AGPL), or the MPLv2.
+
+* All contributions to the project source code ("patches") SHALL use the same license as the project.
+
+* All patches are owned by their authors. There SHALL NOT be any copyright assignment process.
+
+* The copyrights in the project SHALL be owned collectively by all its Contributors.
+
+* Each Contributor SHALL be responsible for identifying themselves in the project Contributor list.
+
++++ Patch Requirements
+
+* Maintainers and Contributors MUST have a Platform account and SHOULD use their real names or a well-known alias.
+
+* A patch SHOULD be a minimal and accurate answer to exactly one identified and agreed problem.
+
+* A patch MUST adhere to the code style guidelines of the project if these are defined.
+
+* A patch MUST adhere to the "Evolution of Public Contracts" guidelines defined below.
+
+* A patch SHALL NOT include non-trivial code from other projects unless the Contributor is the original author of that code.
+
+* A patch MUST compile cleanly and pass project self-tests on at least the principle target platform.
+
+* A patch commit message SHOULD consist of a single short (less than 50 character) line summarizing the change, optionally followed by a blank line and then a more thorough description.
```

Comment:
it become almost standard by some contributors and maintainers to describe the problem in the commit message, I don't think it is a MUST but we may want to write something about that.